dev/generate_rule_templates.py

```diff
@@ -0,0 +1,115 @@
+import argparse
+import os
+import yaml
+import json
+import common
+from pathlib import Path
+
+INTEGRATION_RULE_TEMPLATE_DIR = "../../integrations/packages/cloud_security_posture/kibana/csp_rule_template/"
+
+
+def generate_rule_templates(benchmark: str, selected_rules: list, rule_template_dir: str):
+    """
+    Generate rule templates from existing rules
+    """
+    rules_templates = []
+
+    benchmark_rules_dir = f"../bundle/compliance/{benchmark}/rules/"
+    rules = os.listdir(benchmark_rules_dir)
+    for rule in rules:
+        if rule not in selected_rules and rule.removeprefix("cis_").replace("_", ".") not in selected_rules:
+            continue
+
+        with open(os.path.join(benchmark_rules_dir, f"{rule}/data.yaml"), "r", encoding="utf-8") as f:
+            rule_obj = yaml.safe_load(f.read())["metadata"]
+            rule_obj["rego_rule_id"] = rule
+
+            rule_template = migrate_csp_rule_metadata(
+                {
+                    "id": rule_obj["id"],
+                    "type": "csp-rule-template",
+                    "attributes": rule_obj
+                }
+            )
+
+            rules_templates.append(rule_template)
+
+    # Write templates into file
+    save_rule_templates(rules_templates, rule_template_dir)
+
+
+def save_rule_templates(rule_templates: list[dict], rule_template_dir: str):
+    """
+    Save rule templates to file
+    """
+    if not rule_template_dir:
+        rule_template_dir = INTEGRATION_RULE_TEMPLATE_DIR
+
+    # ensure path exist, else create it
+    Path(rule_template_dir).mkdir(parents=True, exist_ok=True)
+
+    for rule_template in rule_templates:
+        with open(os.path.join(rule_template_dir, f"{rule_template['id']}.json"), "w") as f:
+            json.dump(rule_template, f, indent=4)
+
+
+def migrate_csp_rule_metadata(doc: dict) -> dict:
+    """
+    Migrate rule metadata to integration format
+    """
+    attributes = doc["attributes"]
+    print(attributes.get("rule_number"))
+    metadata = {
+        "impact": attributes.pop("impact", None),
```

Comment:
Up to you 💪 

I just think it looks a bit awkward that sometimes we present `"'` and sometimes we present "None"

Reply:
where are presenting `None`?

Reply:
We talked about it in person.

It was suggested, that we don't feel comfortable with replacing `""` with None, since we cannot commit that the impact is none.

There is a ticket to remove "impact" where the value equals to `""`.